### PR TITLE
Implement CryptoSettings for Linux using libsecret

### PR DESCRIPTION
**Pull request by oskirby**
## Description
Linux devices can make use of the `org.freedesktop.secret` D-Bus API to store and load keys and credentials in a way that protects them against readout without the user's login credentials.

## Reference
Github: #2362 ([VPN-1397](https://mozilla-hub.atlassian.net/browse/VPN-1397))

## Checklist
    
- [x] My code follows the style guidelines for this project
- [x] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [x] I have performed a self review of my own code
- [x] I have commented my code PARTICULARLY in hard to understand areas
- [x] I have added thorough tests where needed

### CODE REVIEW

File: src/platforms/linux/linuxcryptosettings.cpp [hunk 1]
```diff
@@ -3,15 +3,90 @@
...
+      
+      QString b64key(s_key.toBase64());
+      gboolean ok = secret_password_store_sync(cryptosettings_get_schema(),
+                                               SECRET_COLLECTION_DEFAULT,
+                                               "VPN settings encryption key",
+                                               qPrintable(b64key), nullptr,
+                                               &error, nullptr);
+      if (error != nullptr) {
+        logger.error() << "Key storage failed:" << error->message;
+        g_error_free(error);
```
Comment (bakulf) [inline src/platforms/linux/linuxcryptosettings.cpp:75]: return false?

File: src/platforms/linux/linuxcryptosettings.cpp [hunk 2]
```diff
@@ -3,15 +3,90 @@
...
+  }
+
+  logger.warning() << "Invalid key";
   return false;
 }
 
 // static
 CryptoSettings::Version CryptoSettings::getSupportedVersion() {
-  return CryptoSettings::NoEncryption;
+  return CryptoSettings::EncryptionChachaPolyV1;
```
Comment (bakulf) [inline src/platforms/linux/linuxcryptosettings.cpp:91]: is there a way to check the libsecret status?

**Review by bakulf — COMMENTED** [review]
The documentation of libsecret suggests using the _sync methods on non-ui threads.
Do you think it's doable?

File: src/platforms/linux/linuxcryptosettings.cpp (hunk 2 above)
Comment (oskirby) [reply src/platforms/linux/linuxcryptosettings.cpp:91]: Not in the official API as far as I can see, but we could check to see if the `org.freedesktop.secrets` name has been taken on the session D-Bus.

**Comment by oskirby** [comment]
> The documentation of libsecret suggests using the _sync methods on non-ui threads. Do you think it's doable?

Unfortunately, I don't think so. The caller of `CryptoSettings::getKey()` is expecting the key to be ready upon return, which requires us to block until the D-Bus response arrives anyways. Which is precisely what the `_sync()` methods do.

File: src/platforms/linux/linuxcryptosettings.cpp (hunk 2 above)
Comment (bakulf) [reply src/platforms/linux/linuxcryptosettings.cpp:93]: yes, probably it's better. What do you think?

Comment (oskirby) [reply src/platforms/linux/linuxcryptosettings.cpp]: I will add a check to ensure that `org.freedesktop.secrets` exists.

**Review by bakulf — APPROVED** [review]

**Pull request merged**